### PR TITLE
api: Updated the CRD definition

### PR DESCRIPTION
**Pull request by EdDev**

### CODE REVIEW

File: PROJECT [hunk 1]
```diff
@@ -7,12 +7,12 @@ projectName: selfserviceoverlay
...
 - api:
-    crdVersion: v1
+    crdVersion: v1alpha1
     namespaced: true
   controller: true
-  domain: ovn.org
-  group: self.service
-  kind: OverlayNetwork
-  path: github.com/AlonaKaplan/selfserviceoverlay/api/v1
+  domain: kubevirt.org
```
Comment (AlonaKaplan) [inline PROJECT:13]: why kubevirt? This is not a kubevirt project.

File: api/v1alpha1/overlaynetwork_types.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// NetworkAttachmentDefinitionRequestSpec defines the desired state of NetworkAttachmentDefinitionRequest
+type NetworkAttachmentDefinitionRequestSpec struct {
+	Name               string `json:"name"`
+	LogicalNetworkName string `json:"logicalNetworkName"`
+	Type               string `json:"type,omitempty"`
```
Comment (AlonaKaplan) [inline api/v1alpha1/overlaynetwork_types.go:29]: Doesn't the type always - `"ovn-k8s-cni-overlay"`? It is not something we want the user to define.

File: api/v1alpha1/overlaynetwork_types.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Important: Run "make" to regenerate code after modifying this file
+
+// NetworkAttachmentDefinitionRequestSpec defines the desired state of NetworkAttachmentDefinitionRequest
+type NetworkAttachmentDefinitionRequestSpec struct {
```
Comment (AlonaKaplan) [inline api/v1alpha1/overlaynetwork_types.go:26]: Why MTU is not here?

File: api/v1alpha1/overlaynetwork_types.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// NetworkAttachmentDefinitionRequest is the Schema for the network attachment definition request API
+type NetworkAttachmentDefinitionRequest struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NetworkAttachmentDefinitionRequestSpec   `json:"spec,omitempty"`
+	Status NetworkAttachmentDefinitionRequestStatus `json:"status,omitempty"`
```
Comment (AlonaKaplan) [inline api/v1alpha1/overlaynetwork_types.go:45]: Why to keep an empty status?

File: api/v1alpha1/overlaynetwork_types.go [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+//+kubebuilder:object:root=true
+
+// NetworkAttachmentDefinitionRequestList contains a list of NetworkAttachmentDefinitionRequest
+type NetworkAttachmentDefinitionRequestList struct {
```
Comment (AlonaKaplan) [inline api/v1alpha1/overlaynetwork_types.go:51]: Why do we need this list?

File: main.go [hunk 6]
```diff
@@ -30,7 +30,7 @@ import (
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
 
-	selfservicev1 "github.com/AlonaKaplan/selfserviceoverlay/api/v1"
+	selfservicev1 "github.com/AlonaKaplan/selfserviceoverlay/api/v1alpha1"
```
Comment (AlonaKaplan) [inline main.go]: selfservicev1alpha

File: PROJECT [hunk 7]
```diff
@@ -7,12 +7,12 @@ projectName: selfserviceoverlay
 repo: github.com/AlonaKaplan/selfserviceoverlay
 resources:
 - api:
-    crdVersion: v1
+    crdVersion: v1alpha1
```
Comment (AlonaKaplan) [inline PROJECT:10]: Some more places I found v1 version that may need to be changed as well.

config/manager/manager.yaml:apiVersion: v1
config/rbac/service_account.yaml:apiVersion: v1
config/samples/kustomization.yaml:- self.service_v1_overlaynetwork.yaml
config/samples/self.service_v1_overlaynetwork.yaml:apiVersion: self.service.ovn.org/v1

Comment (EdDev) [reply PROJECT:10]: Who generated those?

File: PROJECT (hunk 1 above)
Comment (EdDev) [reply PROJECT:13]: We need a "sponsor", this is the closest one that I know and can let this project live under.

File: api/v1alpha1/overlaynetwork_types.go (hunk 5 above)
Comment (ormergi) [reply api/v1alpha1/overlaynetwork_types.go:51]: I think it enables listing the CRD instances, for example `kubectl get <crd-name>`

File: api/v1alpha1/overlaynetwork_types.go (hunk 3 above)
Comment (EdDev) [reply api/v1alpha1/overlaynetwork_types.go:26]: There are two ways to go forward:
- Add specific fields which we will support, tracking a specific CNI.
- Add the base, then support generic fields per configuration. This aims for a generic solution.

I think we can start with the minimum, which is the name of the NAD, logical-network and the type (see below why).

File: api/v1alpha1/overlaynetwork_types.go (hunk 2 above)
Comment (ormergi) [inline api/v1alpha1/overlaynetwork_types.go:29]: Could you please explain what each filed represent and why we need it?

File: api/v1alpha1/overlaynetwork_types.go (hunk 3 above)
Comment (ormergi) [inline api/v1alpha1/overlaynetwork_types.go:26]: It seems that this definition enable more granular settings for the NAD although at this point the controller should only care for overlay-network settings.

Where could we set the concrete NAD settings for overlay this controller should eventually manage?

File: api/v1alpha1/overlaynetwork_types.go (hunk 2 above)
Comment (EdDev) [reply api/v1alpha1/overlaynetwork_types.go:29]: I would prefer we define it explicitly, to make this a generic solution that will fit other CNIs.

It is optional, so you do not have to use it and set in the code a default. (not that I want this for production)

File: api/v1alpha1/overlaynetwork_types.go (hunk 3 above)
Comment (ormergi) [reply api/v1alpha1/overlaynetwork_types.go:26]: The NAD CR is already long and cumbersome to work with, can we think of shorter name?
Maybe network-request, network-setting?

File: api/v1alpha1/overlaynetwork_types.go (hunk 4 above)
Comment (EdDev) [reply api/v1alpha1/overlaynetwork_types.go:45]: We need to fill it later, I do not think it needs to stay empty.
I also find it odd to see this gone, but I guess we could get rid of it for now.

File: api/v1alpha1/overlaynetwork_types.go (hunk 5 above)
Comment (EdDev) [reply api/v1alpha1/overlaynetwork_types.go:51]: It was auto-generated before, I guess it serves listing the objects of this type?

File: main.go (hunk 6 above)
Comment (EdDev) [reply main.go]: I'm not even sure why it is aliased.
Done

File: PROJECT (hunk 7 above)
Comment (AlonaKaplan) [reply PROJECT:10]: I think it was part of the project initialization. Now it has to be updated manually.

File: api/v1alpha1/overlaynetwork_types.go (hunk 5 above)
Comment (AlonaKaplan) [reply api/v1alpha1/overlaynetwork_types.go:51]: Yes, it was auto-generated. But maybe we don't need it. Would watching the CRD work without it?

File: api/v1alpha1/overlaynetwork_types.go (hunk 3 above)
Comment (AlonaKaplan) [reply api/v1alpha1/overlaynetwork_types.go:26]: We need to decide whether we choose a specific or the general approach.
The project was created specifically for overlay network. The name of the file is overlaynetwork_type. And we have overlay_controller. Overlay is all over the project.
I would prefer to keep it specific for the POC.
Later, if this POC will become a real project we can consider writing it in a generic way,

Comment (phoracek) [reply api/v1alpha1/overlaynetwork_types.go:26]: We want a PoC. Sooner we get something that presents the basic concept of self-service overlay networks the better. This repo should be treated as something that might as well be thrown to trash the moment we finish the PoC. The concept we are trying to prove and present are self-service networks, **not** a generic NAD operator. Please make an effort focusing on the minimal set of features we need for the PoC. Spending time thinking on any stretch goals is making the PoC worse.

**Review by EdDev — COMMENTED** [review]
Per the feedback, I understand that this change is not needed.
For a PoC, we do not need the version to be correct, nor "better" CRD name or generalization of the whole solution.

I'll open a different PR that just adds some minimal fields to serve the ovn-kubernetes CNI.

/close

File: api/v1alpha1/overlaynetwork_types.go (hunk 5 above)
Comment (EdDev) [reply api/v1alpha1/overlaynetwork_types.go:51]: It is needed for listing all CR/s of a specific CRD, which sounds as a requirement for an informer.

**Pull request closed without merging**